src/diff/index.js

```diff
@@ -365,10 +367,22 @@ function doRender(props, state, context) {
  * component check for error boundary behaviors
  */
 function catchErrorInComponent(error, component) {
+	// thrown Promises are meant to suspend...
+	let isSuspend = typeof error.then === 'function';
+	let suspendingComponent = component;
+
 	for (; component; component = component._ancestorComponent) {
 		if (!component._processingException) {
 			try {
-				if (component.constructor.getDerivedStateFromError!=null) {
+				if (isSuspend) {
+					if (component._childDidSuspend) {
+						component._childDidSuspend(error);
+					}
+					else {
+						continue;
```

Comment:
Maybe move to here?

```suggestion
		return catchErrorInComponent(Error('Missing Suspense'), suspendingComponent);
```

Reply:
The `continue` results in traversing up the tree. This would break the tree traversing AFAIK